### PR TITLE
fix(gzip)!: change the default block size

### PR DESCRIPTION
**Pull request by rchincha**
BREAKING CHANGE: the default gzip block size is changed to 256<<12, was previously 256<<10.

The layers created by ecosystem tools and stacker are different for the same tar content. Unfortunately, the OCI image spec doesn't standardize/encode this in the document. Hence, we standardize in the implementation.

We now link against our own fork: github.com/project-stacker/umoci which may change depending on the PR getting merged to upstream.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/overlay/pack.go [hunk 1]
```diff
@@ -408,7 +408,7 @@ func generateLayer(config types.StackerConfig, oci casext.Engine, mutators []*mu
 		defer blob.Close()
 
 		if layerType.Type == "tar" {
-			desc, err = mutator.Add(context.Background(), mediaType, blob, history, mutate.GzipCompressor, nil)
+			desc, err = mutator.Add(context.Background(), mediaType, blob, history, mutate.GzipCompressor.WithOpt(mutate.GzipBlockSize(256<<12)), nil)
```
Comment (raharper) [inline pkg/overlay/pack.go:411]: Could we move the size to a const somewhere that's searchable, also with a comment indicating what it was previously and why we're changing it now?

Comment (rchincha) [reply pkg/overlay/pack.go:411]: The commit msg/log captures that. But I can add a note.
Fixing the const.

Comment (rchincha) [reply pkg/overlay/pack.go]: Fixed.

**Review by raharper — COMMENTED** [review]
Suggestion in commit message:

A tar layer with the same content but compressed with different gzip blocksize
will result in different sha256sums in the final OCI Image.  Ecosystem tools
have one current size in use and stacker's current size differ.

Interactions between a stacker-built OCI image and ecosystem tools
which *recompress* lower layers results in bloated registries which will have
identical tar content but different compressed sha256 blobs.

Unfortunately, the OCI image spec doesn't standardize/encode this in the
specification document. Hence, we change to the current common block size used
in the ecosystem here in the stacker implementation.

**Comment by rchincha** [comment]
> Suggestion in commit message:

Updated.

**Review by raharper — APPROVED** [review]

**Pull request merged**